src/transactions/TransactionFrame.cpp

```diff
@@ -1352,6 +1355,24 @@ TransactionFrame::processFeeSeqNum(AbstractLedgerTxn& ltx,
     }
 }
 
+bool
+TransactionFrame::isValidStructure() const
```

Comment:
nit: I think `isValidStructure` is a bit misleading given that this is a required invariant for fee-related processing only. So maybe rename this to reflect that better, like `isValidForFeeProcessing` or something like that. Maybe could add some negative fee protection in fee bump envelope as well. Up to you though - if you want to put non-fee-related checks here (but it seems like that would dilute the validation logic a bit too much IMHO)